src/main/java/algorithms/hard/MaximumSumof3NonOverlappingSubarrays.java

```diff
@@ -0,0 +1,57 @@
+package algorithms.hard;
+
+public class MaximumSumof3NonOverlappingSubarrays {
+
+    public int[] maxSumOfThreeSubarrays(int[] nums, int k) {
+        int n = nums.length;
+        int[] sum = new int[n + 1], leftBestIdx = new int[n], rightBestIdx = new int[n];
+
+        for (int i = 0; i < n; i++) {
+            sum[i + 1] = sum[i] + nums[i];
+        }
+
+        for (int i = k, maxSum = sum[k] - sum[0]; i < n; i++) {
+            if (sum[i + 1] - sum[i + 1 - k] > maxSum) {
```

Comment:
Strictly greater because of lexographical order. We need to stay left if the sums are same.